search/django/weedid/utils.py

```diff
@@ -118,3 +147,20 @@ def retrieve_listing_info(query_entity, awaiting_review):
         "contributor": query_entity.user.username,
         "contributor_email": query_entity.user.email if awaiting_review else "",
     }
+
+
+def retrieve_category_name(category):
+    if re.fullmatch(r"(crop|weed): ([a-z][a-z ]+|UNSPECIFIED)", category["name"]):
+        return {
+            "id": category["id"],
+            "category": category["name"].split(": ")[0],
+            "name": category["name"].split(": ")[1],
```

Comment:
shouldn't this be `category["name"]`?

Reply:
I'm differentiating 2 scenarios that when the name already matches with the pattern `r"(crop|weed): ([a-z][a-z ]+|UNSPECIFIED)"`, I extract category and name from it and give the frontend a preset of the current category but also let them change it.